### PR TITLE
implement client.

### PR DESCRIPTION
**Pull request by MingzhouLin**
I implement the client class. And make ToDo comments on that.

### CODE REVIEW

File: src/CenterSystem.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import java.beans.BeanInfo;
+import java.beans.*;
```
Comment (Tumerskiy) [inline src/CenterSystem.java:1]: bad practice

File: src/CenterSystem.java [hunk 2]
```diff
@@ -37,7 +34,7 @@ public void setCenterName(String centerName) {
         this.centerName = centerName;
     }
 
-    public void createTRecord(String firstName, String lastName, String address, int phone, String specialization, String location) throws RemoteException {
+    public void createTRecord(String managerId, String firstName, String lastName, String address, int phone, String specialization, String location) throws RemoteException {
```
Comment (Tumerskiy) [inline src/CenterSystem.java]: I thought you were willing to get rid of this argument

File: src/CenterSystem.java (hunk 1 above)
Comment (MingzhouLin) [reply src/CenterSystem.java:1]: I didn't change it ..... maybe some modification in our code causes it

File: src/Client.java [hunk 3]
```diff
@@ -21,5 +22,141 @@ public static void main(String args[]) throws Exception{
...
+        Registry registry = LocateRegistry.getRegistry();
+        CenterServer stub;
+        String managerId = "";
+        boolean ifContinue = false;
+        do {
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("Please input your manager ID:");
+            managerId = scanner.nextLine();
+            verifyId(managerId);
+            switch (managerId.substring(0, 3)) {
```
Comment (Tumerskiy) [inline src/Client.java]: what about default case?

Comment (MingzhouLin) [reply src/Client.java]: Default case? like what?

File: src/Client.java [hunk 4]
```diff
@@ -21,5 +22,141 @@ public static void main(String args[]) throws Exception{
...
+                case "DDO": {
+                    stub = (CenterServer) registry.lookup("DDO");
+                    ifContinue = processOpt(stub);
+                    break;
+                }
+            }
+        } while (ifContinue);
+    }
+
+    public boolean processOpt(CenterServer stub) throws Exception {
```
Comment (Tumerskiy) [inline src/Client.java]: looks heavy :)
you do the "validation" of request on client side, using this master... We have it on server side also, we may go with easier way on client side also, but still, seems good idea to have it like this

File: src/Client.java [hunk 5]
```diff
@@ -21,5 +22,141 @@ public static void main(String args[]) throws Exception{
...
+        System.out.println("Please input your teacher's specialization:");
+        teacher.addSpecialiazation(scanner.nextLine().trim());
+        System.out.println("Please input teacher's location:");
+        teacher.setLocation(scanner.nextLine().trim());
+        System.out.println("Please input teacher's phone:");
+        teacher.setPhone(scanner.nextInt());
+        stub.createTRecord(teacher.getFirstName(), teacher.getLastName(), teacher.getAddress(), teacher.getPhone(), teacher.getSpecialiazation(), teacher.getLocation());
+    }
+
+    public void processOpt2(CenterServer stub) throws RemoteException {
```
Comment (Tumerskiy) [inline src/Client.java]: try to give meaningful names, otherwise code is a bit cryptic

File: src/Client.java [hunk 6]
```diff
@@ -21,5 +22,141 @@ public static void main(String args[]) throws Exception{
...
+        System.out.println("Please input student's courses(split with space):");
+        for (String s :
+                scanner.nextLine().split(" ")) {
+            student.getCoursesRegistered().add(s);
+        }
+        stub.createSRecord(student.getFirstName(), student.getLastName(), student.getCoursesRegistered(), student.getStatus(), student.getStatusDate());
+    }
+
+    public int processOpt3(CenterServer stub) {
+        //TODO: We should let this function return int and remove the argument "managerId".
```
Comment (Tumerskiy) [inline src/Client.java]: Why? 
expected format should contain server name and number of records, not 1 int,
so I suggest MTL:1 LVL:2 DDO:3... as the way to go format

Comment (MingzhouLin) [reply src/Client.java]: Make sense. I will remove it

File: src/Client.java [hunk 7]
```diff
@@ -21,5 +22,141 @@ public static void main(String args[]) throws Exception{
...
+        String recordId = scanner.nextLine().trim();
+        System.out.println("Please input the field name you want to change:");
+        String fieldName = scanner.nextLine().trim();
+        System.out.println("Please input new value:");
+        String newValue = scanner.nextLine().trim();
+        stub.editRecord(recordId, fieldName, newValue);
+    }
+
+
+    public void verifyId(String managerId) throws Exception {
```
Comment (Tumerskiy) [inline src/Client.java]: what is it for? we should address this validation with case in the beginning right?

File: src/Client.java (hunk 3 above)
Comment (Tumerskiy) [reply src/Client.java]: retrun "wrong login, try again"?

**Review by Tumerskiy — CHANGES_REQUESTED** [review]
some little touch ups are required

**Pull request merged**